pkgs/by-name/lo/logseq/package.nix

```diff
@@ -0,0 +1,286 @@
+        exec ${lib.getExe' clojure "clojure"} -Sdeps '{:mvn/local-repo "${finalAttrs.mavenRepo}"}' "$@"
+      '';
+
+      fakeGit = writeShellScriptBin "git" ''
+        echo "${finalAttrs.src.rev or finalAttrs.version}@nixpkgs"
+      '';
+    in
+    [
+      yarnConfigHook
+      yarnBuildHook
+      nodejs
+      (nodejs.python.withPackages (ps: [ ps.setuptools ]))
+      pkg-config
+      clojureWithCache
+      fakeGit
+      zip
+      makeWrapper
+      removeReferencesTo
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      xcbuild
+      cctools
+    ];
+
+  buildInputs = [
+    # deps for the canvas npm dependency
+    pixman
+    cairo
+    pango
+  ];
+
+  # we'll run the hook manually multiple times
+  dontYarnInstallDeps = true;
+
+  env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+  postConfigure = ''
+    yarnOfflineCache="$yarnOfflineCacheRoot" yarnConfigHook
+
+    cp resources/package.json static/package.json
+    pushd static
+    yarnOfflineCache="$yarnOfflineCacheResources" yarnConfigHook
+    popd
+
+    pushd packages/amplify
+    yarnOfflineCache="$yarnOfflineCacheAmplify" yarnConfigHook
+    popd
+
+    pushd tldraw
+    cp yarn.lock apps/tldraw-logseq/yarn.lock
+    yarnOfflineCache="$yarnOfflineCacheTldraw" yarnConfigHook
+    pushd apps/tldraw-logseq
+    yarnOfflineCache="$yarnOfflineCacheTldraw" yarnConfigHook
+    popd
+    popd
+
+    # this has to be done after everything is set up, because for some reason
+    # the shebangs somehow get unpatched... I don't know why...
+    patchShebangs node_modules
+    patchShebangs static/node_modules
+    patchShebangs packages/amplify/node_modules
+    patchShebangs tldraw/node_modules
+    patchShebangs tldraw/apps/tldraw-logseq/node_modules
+
+    yarn --offline --cwd tldraw postinstall
+    yarn --offline --cwd packages/amplify postinstall
+
+    export npm_config_nodedir=${electron-headers}
+
+    pushd static
+
+    # we want to use our own git, don't try downloading it
+    substituteInPlace node_modules/dugite/package.json \
+      --replace-fail '"postinstall"' '"_postinstall"'
+
+    # this doesn't seem to build with electron-headers
+    rm node_modules/macos-alias/binding.gyp
+
+    # the electron-rebuild deadlocks for some reason, let's just use normal npm rebuild (since we overrode the nodedir anyways)
+    npm rebuild --verbose
+
+    # remove references to the set nodedir
+    find node_modules -type f \( -name "*.target.mk" -o -name "config.gypi" -o -name "Makefile" \) -delete
+
+    popd
+
+    cp -r ${electron.dist} electron-dist
+    chmod -R u+w electron-dist
+
+    pushd electron-dist
+    zip -0Xqr ../electron.zip .
+    popd
+
+    rm -r electron-dist
+
+    substituteInPlace static/node_modules/@electron/packager/dist/packager.js \
+      --replace-fail "await this.getElectronZipPath(downloadOpts)" "\"$(pwd)/electron.zip\""
+
+    cp -r static/node_modules resources/node_modules
+  '';
+
+  yarnBuildScript = "release-electron";
+
```

Comment:
```suggestion

  desktopItem = makeDesktopItem {
    name = "com.logseq.Logseq";
    exec = "logseq %U";
    terminal = false;
    icon = "com.logseq.Logseq";
    type = "Application";
    comment = "A privacy-first, open-source platform for knowledge management and collaboration.";
    desktopName = "Logseq";
    startupWMClass = "Logseq";
    mimeTypes = "x-scheme-handler/logseq";
    categories = ["Utility"];
  };
```

Reply:
which should be used? `%U` or `%u`

Reply:
I think `%U` as logseq could probably open multiple files. That seems to be [how the pro's do it](https://github.com/flathub/com.logseq.Logseq/blob/59ee044f16566e56671865aeafd6564b51708d21/com.logseq.Logseq.desktop#L3), at least 😆